spec/dplex-test-spec.js

```diff
@@ -0,0 +1,205 @@
+
+describe('InvertedIndex class', () => {
+  beforeEach(() => {
+    this.indexInstance = new InvertedIndex();
+    this.validBook = [{ title: 'Welcome to Test Environment',
+      text: 'Enjoy this file' }];
+    this.books = [
+      {
+        title: 'Alice in Wonderland',
+        text: 'Alice falls into a rabbit hole and enters a world full of imagination.'
+      },
+
+      {
+        title: 'The Lord of the Rings: The Fellowship of the Ring.',
+        text: 'An unusual alliance of man, elf, dwarf, wizard and hobbit seek to destroy a powerful ring.'
+      },
+      {
+        title: 'The Lord of the Rings: The Fellowship of the Ring.',
+        text: 'An unusual alliance of man, elf, dwarf, wizard and hobbit seek to destroy a powerful ring.'
+      }
+    ];
+  });
+  describe('InvertedIndex class', () => {
+    it('should check that the class has a createIndex method', () => {
+      expect(typeof this.indexInstance.createIndex).toBe('function');
+    });
+
+    it('should check that the class has a readFile method', () => {
+      expect(typeof this.indexInstance.readFile).toBe('function');
+    });
+
+    it('should check that the class has a validateFile method', () => {
+      expect(typeof this.indexInstance.validateFile).toBe('function');
+    });
+
+    it('should check that the class has a tokenize method', () => {
+      expect(typeof this.indexInstance.tokenize).toBe('function');
+    });
+
+    it('should check that the class has a getDocumentTokens method', () => {
+      expect(typeof this.indexInstance.getDocumentTokens).toBe('function');
+    });
+
+    it('should check that the class has a getIndex method', () => {
+      expect(typeof this.indexInstance.getIndex).toBe('function');
+    });
+
+    it('should check that the class has a searchIndex method', () => {
+      expect(typeof this.indexInstance.searchIndex).toBe('function');
+    });
+
+    it('should check that the class has a getSearchResults method', () => {
+      expect(typeof this.indexInstance.getSearchResults).toBe('function');
+    });
+
+    it('should check that the class has a getDocuments method', () => {
+      expect(typeof this.indexInstance.getDocuments).toBe('function');
+    });
+  });
+
+  describe('Create Index', () => {
+    it('should return mapped indices to words in a JSON file', () => {
```

Comment:
Did you test for when the book was not successfully created? Or do you expect `createIndex` to always succeed?